### PR TITLE
fix(dashboardeditor): need to refactor function to react component so it can be memoized

### PR DESCRIPTION
**Pull request by scottdickerson**
Need to add lots of memoizing to React Editor components to improve performance

Closes #

**Summary**

- fix(BarChartCard): memoize the sample data creation
- fix(DashboardEditor): wrap all function definitions with useCallback and memoize where possible
- refactor(DashboardEditorCardRenderer): need to change the getCardPreview to a React component so I have more control over memoizing and rerender cycle
- refactor(editorUtils): move the render helper functions to DashboardEditorCardRenderer, ideally they will be all migrated to components rather than render functions

**Acceptance Test (how to verify the PR)**

- Verify the DashboardEditor story

### CODE REVIEW

File: src/components/DashboardEditor/DashboardEditor.jsx [hunk 1]
```diff
@@ -379,34 +389,79 @@ const DashboardEditor = ({
...
-    onBrowseClick:
-      cardConfig.type === CARD_TYPES.IMAGE && isNil(cardConfig.content?.src)
-        ? handleShowImageGallery
-        : undefined,
-    validateUploadedImage:
-      cardConfig.type === CARD_TYPES.IMAGE
-        ? onValidateUploadedImage
-        : undefined,
-  });
+  const commonCardProps = useCallback(
```
Comment (bjornalm) [inline src/components/DashboardEditor/DashboardEditor.jsx:392]: Not part of this PR, but I think we should change the name to `getCommonCardProps` or something to indicate that this is a function.

File: src/components/DashboardEditor/DashboardEditor.jsx [hunk 2]
```diff
@@ -379,34 +389,79 @@ const DashboardEditor = ({
...
+          renderCardPreview(
+            cardConfig,
+            cardProps,
+            setSelectedCardId,
+            duplicateCard,
+            removeCard,
+            isSelected,
+            handleShowImageGallery
+          ) ?? (
+            <DashboardEditorCardRenderer
```
Comment (bjornalm) [inline src/components/DashboardEditor/DashboardEditor.jsx:443]: nice

File: src/components/DashboardEditor/DashboardEditorCardRenderer.test.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { CARD_SIZES, CARD_TYPES } from '../../constants/LayoutConstants';
+
+import DashboardEditorCardRenderer from './DashboardEditorCardRenderer';
+
+describe('DashboardEditorCardRenderer', () => {
```
Comment (bjornalm) [inline src/components/DashboardEditor/DashboardEditorCardRenderer.test.jsx:8]: Should we not make sure that all types of cards are rendered in this test file?

**Review by bjornalm — APPROVED** [review]
Nice work, two comments to fix if you have the time.

File: src/components/DashboardEditor/DashboardEditorCardRenderer.test.jsx (hunk 3 above)
Comment (scottdickerson) [reply src/components/DashboardEditor/DashboardEditorCardRenderer.test.jsx:8]: ideally we should but the other kinds of cards are covered by snapshot testing, this was really just added to help with coverage
